x-pack/plugins/security/server/lib/authentication/authentication_result.ts

```diff
@@ -6,183 +6,158 @@
 
 /**
  * Represents status that `AuthenticationResult` can be in.
- * @enum {string}
  */
-const AuthenticationResultStatus = Object.freeze({
+enum AuthenticationResultStatus {
   /**
    * Authentication of the user can't be handled (e.g. supported credentials
    * are not provided).
    */
-  NotHandled: 'not-handled',
+  NotHandled = 'not-handled',
 
   /**
    * User has been successfully authenticated. Result should be complemented
    * with retrieved user information and optionally with state.
    */
-  Succeeded: 'succeeded',
+  Succeeded = 'succeeded',
 
   /**
    * User can't be authenticated with the provided credentials. Result should
    * include the error that describes the reason of failure.
    */
-  Failed: 'failed',
+  Failed = 'failed',
 
   /**
    * Authentication consists of multiple steps and user should be redirected to
    * a different location to complete it. Can be complemented with optional state.
    */
-  Redirected: 'redirected'
-});
+  Redirected = 'redirected',
+}
+
+/**
+ * Represents additional authentication options.
+ */
+interface AuthenticationOptions {
+  error?: Error;
+  redirectURL?: string;
+  state?: unknown;
```

Comment:
__note:__ Since we don't have _yet_ a shape for `user` (I'm thinking about introducing one though) and `state` is a black box I thought having `unknown` here is the safest option.